test/matrix.lean

```diff
@@ -0,0 +1,169 @@
+/-
+manually ported from
+https://github.com/leanprover-community/mathlib/blob/4f4a1c875d0baa92ab5d92f3fb1bb258ad9f3e5b/test/matrix.lean
+-/
+import Mathlib.Data.Matrix.Notation
+-- import linear_algebra.Matrix.determinant
+import Mathlib.GroupTheory.Perm.Fin
+-- import Mathlib.Tactic.NormSwap
+import Qq
+
+open Qq
+
+-- todo: uncomment above imports when they are ported
+
+variables {α β : Type} [Semiring α] [Ring β]
+
+namespace Matrix
+
+open Matrix
+
+/-! Test that the dimensions are inferred correctly, even for empty matrices -/
+section dimensions
+
+-- set_option pp.universes true
+-- set_option pp.all true
+
+section elaborators
+open Lean Meta Elab Command
+
+/-- `dims% e` elaborates `e` as a Matrix and returns its dimensions as a `Nat × Nat`. -/
```

Comment:
This isn't testing the right thing any more; the original was intended to test that the dimensions were syntactically `2` and not `succ (succ 0)` or similar.

Reply:
Ah, perhaps the docstring is just wrong; looking more closely it looks like what you have is testing the right thing after all.

Reply:
No, it is. The `#guard` command is checking syntactic equality.

```
#guard dims% !![;;]      is (2, 0) -- succeeds
#guard dims% !![;;]      is (Nat.succ 1, 0) -- fails
```